sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/ReadResource.java

```diff
@@ -0,0 +1,400 @@
+        content = {
+          @Content(
+              mediaType = MediaType.APPLICATION_JSON,
+              schema =
+                  @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                      name = "SearchDocsResult",
+                      properties = {
+                        @SchemaProperty(
+                            name = "data",
+                            type = SchemaType.ARRAY,
+                            minItems = 0,
+                            example = "[]"),
+                        @SchemaProperty(
+                            name = "pageState",
+                            type = SchemaType.STRING,
+                            example = "c29tZS1leGFtcGxlLXN0YXRl"),
+                        @SchemaProperty(name = "profile", implementation = ExecutionProfile.class)
+                      }),
+              examples = {
+                @ExampleObject(ref = OpenApiConstants.Examples.SEARCH_DOCUMENTS),
+                @ExampleObject(ref = OpenApiConstants.Examples.SEARCH_DOCUMENTS_UNWRAPPED)
+              })
+        }),
+    @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+    @APIResponse(
+        responseCode = "404",
+        description = "Not found.",
+        content =
+            @Content(
+                examples = {
+                  @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST),
+                  @ExampleObject(ref = OpenApiConstants.Examples.COLLECTION_DOES_NOT_EXIST)
+                },
+                schema =
+                    @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                        implementation = ApiError.class))),
+    @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+    @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+    @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+  })
+  @GET
+  public Uni<RestResponse<Object>> searchDocuments(
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @QueryParam("where") String where,
+      @QueryParam("fields") String fields,
+      @QueryParam("page-size")
+          @Min(value = 1, message = "the minimum number of results to return is one")
+          @Max(value = 20, message = "the max number of documents to return is 20")
+          Integer pageSize,
+      @QueryParam("page-state") String pageState,
+      @QueryParam("profile") boolean profile,
+      @QueryParam("raw") boolean raw) {
+
+    // fetch a valid table to ensure read is from a collection table
+    return tableManager
+        .getValidCollectionTable(namespace, collection)
+
+        // if exists, then map to the read action
+        .flatMap(
+            t -> {
+              ExecutionContext context = ExecutionContext.create(profile);
+
+              // default page size of 3
+              int pageSizeFinal = Optional.ofNullable(pageSize).orElse(3);
```

Comment:
This is just my uninformed opinion, but I think increasing makes sense (not surprised there's "only 3?" question every now and then) and somehow feel 10 might be a good default; anything less seems unusual. 20 would work too.

Reply:
Would keep 3 for now, but yes up for the discussion.. Main reason was not to load too many documents on the "default" page size..

Reply:
Right, I remember hearing that the low size was meant to limit disk access and in a way benefit users.
This might be something to talk with Lorina and/or Mark at some point. Maybe it is better to wait for user feedback before changing things.